sphinxcontrib/inmanta/api.py

```diff
@@ -476,21 +644,149 @@ def run(self, module_repo: Optional[str], module_name: str, extra_modules: Seque
+    ),
+)
+@click.option(
+    "--module-name", help="The module to generate api docs for", required=True
+)
+@click.option(
+    "--extra-modules",
+    "-m",
+    help="Extra modules that should be loaded to render the docs",
+    multiple=True,
+)
+@click.option(
+    "--out-dir", "-d", help="Place doc output in this directory.", required=True
+)
+def generate_api_doc_v2(
+    module_repo: Optional[str],
+    module_name: str,
+    extra_modules: Sequence[str],
+    out_dir: str,
+):
+    """
+    Generate API documentation for a module. This command expects the module to live in a directory
+    that has the module's name and that is in the directory from which the command is called.
+
+    The output of this command depends on whether the module contains
+    a README.md file or not:
+
+    If the module contains a README.md file, a doc directory named <module_name> for this module is created in out_dir.
+    This folder is then populated with relevant doc info for this module:
+        - The README.md file
+        - The changelog.md file
+        - The docs folder (if present)
+
+    In addition, if the README.md file references ``autodoc.rst``, then api documentation for this module will be
+    generated and placed in a ``autodoc.rst`` file in the <module_name> directory
+
+    If the module doesn't contain a README.md, api documentation for this module is generated and placed in a
+    <module_name>.rst file in out_dir.
     """
-        Generate API documentation for module
+
+    module_dir = os.path.abspath(os.path.join(os.path.curdir, module_name))
+    readme_file = os.path.join(module_dir, "README.md")
+
+    if not os.path.exists(readme_file):
+        write_auto_doc(
+            extra_modules,
+            module_name,
+            module_repo,
+            os.path.join(os.path.abspath(module_name), AUTODOC_FILE),
+        )
+        return
+
+    module_doc_dir = build_module_doc_directory(out_dir, module_dir, module_name)
+    with open(readme_file, "r") as f:
+        for line in f:
+            if AUTODOC_FILE in line:
+                write_auto_doc(
+                    extra_modules,
+                    module_name,
+                    module_repo,
+                    os.path.join(os.path.abspath(module_doc_dir), AUTODOC_FILE),
+                )
+                break
+
+
+def build_module_doc_directory(out_dir: str, module_dir, module_name: str) -> str:
+    """
+    Create a documentation directory named <module_name> in <out_dir> directory and return the path to it.
+    In addition, relevant documentation files/directories
+
+    :param out_dir: Root dir in which to create the <module_name> directory.
+    :param module_dir: Path to the module.
+    :param module_name: Name of the module doc directory to create.
+    :return: Path to the created directory.
+    """
+    module_doc_dir = os.path.abspath(os.path.join(out_dir, module_name))
+    os.makedirs(module_doc_dir)
+
+    files_to_copy = [
```

Comment:
This list also contains directories.
```suggestion
    to_copy = [
```